### PR TITLE
WT-9753 Create a docker container for WiredTiger

### PR DESCRIPTION
**Pull request by agorrod**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -174,6 +175,9 @@ if(WITH_PIC OR ENABLE_SHARED)
     if(ENABLE_TCMALLOC)
         target_include_directories(wt_objs PRIVATE ${HAVE_LIBTCMALLOC_INCLUDES})
     endif()
+    if(ENABLE_TCMALLOC)
```
Comment (agorrod) [inline CMakeLists.txt:178]: Should this be `ENABLE_VOIDSTAR`?

**Review by agorrod — COMMENTED** [review]
This seems OK to me as-is.
If we decide we want to merge it into WiredTiger permanently I've got some opinions about naming and layering.

File: CMakeLists.txt (hunk 1 above)
Comment (mickgraham-mongodb) [reply CMakeLists.txt]: Fixed

File: cmake/third_party/voidstar.cmake [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+add_library(wt::voidstar INTERFACE IMPORTED GLOBAL)
+set_target_properties(wt::voidstar PROPERTIES
+    IMPORTED_LIBNAME voidstar
+)
+
+link_directories(${CMAKE_SOURCE_DIR}/tools/voidstar/lib)
+
+set_target_properties(wt::voidstar PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_SOURCE_DIR}/tools/voidstar/include
+)
```
Comment (luke-pearson) [inline cmake/third_party/voidstar.cmake:18]: Should be a newline at the end of the file.

File: test/format/CONFIG.antithesis [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# A reasonable configuration for stress testing.
+btree.huffman_value=0
+cache.minimum=20
+runs.rows=100000:500000
+runs.tables=5
+runs.threads=4:32
+runs.timer=5
+runs.type=row,var
```
Comment (luke-pearson) [inline test/format/CONFIG.antithesis:8]: Again newline.

File: cmake/configs/base.cmake [hunk 4]
```diff
@@ -361,3 +367,7 @@ if(WT_WIN)
         add_compile_options(/MT)
     endif()
 endif()
+
+if(ENABLE_ANTITHESIS)
+    add_compile_options(-fsanitize-coverage=trace-pc-guard)
+endif()
```
Comment (luke-pearson) [inline cmake/configs/base.cmake:373]: Again newline

File: DOCKER_README [hunk 5]
```diff
@@ -0,0 +1,9 @@
+This branch is to run format as a docker container to work with Antithesis.
+
+1. Build with these options
+    cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/mongodbtoolchain_v3_clang.cmake -DENABLE_ANTITHESIS=1  -G "Ninja" ./..
+2. Install docker on your host
+3. Execute the following command to build the docker container and run the docker image:
+    sudo docker build -f test_format.dockerfile -t image_name . && sudo docker run image_name
+
+We are including libstarvoid which helps with Antithesus integration.
```
Comment (ddanderson) [inline DOCKER_README:9]: Antithesus -> Antithesis

File: DOCKER_README [hunk 6]
```diff
@@ -0,0 +1,9 @@
+This branch is to run format as a docker container to work with Antithesis.
+
+1. Build with these options
+    cmake -DCMAKE_TOOLCHAIN_FILE=../cmake/toolchains/mongodbtoolchain_v3_clang.cmake -DENABLE_ANTITHESIS=1  -G "Ninja" ./..
```
Comment (ddanderson) [inline DOCKER_README:4]: Do we need an explicit call to "ninja" after this?

**Review by ddanderson — APPROVED** [review]
LGTM, saw some small things in the readme.

**Comment by sauclovian-wt** [comment]
Two kibitzes:
- wouldn't it be better to compile libvoidstar than include a binary blob? Given that it seems to be very small...
- shouldn't the docker container itself compile WT?
Feel free to ignore me, I'm sticking my oar in where it probably doesn't belong :-)

**Comment by mickgraham-mongodb** [comment]
This is a small POC for integration with a 3rd party.

> wouldn't it be better to compile libvoidstar than include a binary blob? Given that it seems to be very small...
We don't have the source at this stage. But yes it would be.
> shouldn't the docker container itself compile WT?
Not for this use case (we only want the running of format, but in the future I do want to use docker more for our builds)

**Comment by sauclovian-wt** [comment]
> This is a small POC for integration with a 3rd party.
Fair enough. Don't mind me, just thought I'd look at it because I've dealt with docker in the past. It can be very convenient; but it also has real costs...

**Comment by agorrod** [comment]
@mickgraham-mongodb - what is the status of this branch? Should we get it into shape to merge, or discard, or just close the PR and leave the branch alive (that's a bit dangerous, since I think it could be auto-pruned at some stage).

**Comment by mickgraham-mongodb** [comment]
> what is the status of this branch? Should we get it into shape to merge, or discard, or just close the PR and leave the branch alive (that's a bit dangerous, since I think it could be auto-pruned at some stage).

We are waiting for feedback from Antithesis on an issue with format not stopping (which looks like an issue on the Antithesis side). I would prefer to keep this as a branch but we could merge it if needed.

**Comment by agorrod** [comment]
@mickgraham-mongodb - is this still alive? Should we get some version of it merged in?

**Comment by mickgraham-mongodb** [comment]
Actively working on rebasing and getting it merged in. Targeting this week (once I work out the version settings we will use in the containers).

**Review by agorrod — COMMENTED** [review]
LGTM - thanks Mick. Though I can't click LGTM, since I raised the PR.

**Comment by mickgraham-mongodb** [comment]
@ddanderson can you please take a look at the changes (more since last time).

File: test/format/ops.c [hunk 7]
```diff
@@ -416,12 +416,10 @@ operations(u_int ops_seconds, u_int run_current, u_int run_total)
         if (!running)
             break;
         __wt_sleep(0, 250 * WT_THOUSAND); /* 1/4th of a second */
-
-        if (fourths == 1 && GV(RUNS_PREDICTABLE_REPLAY))
-            replay_end_timed_run();
```
Comment (ddanderson) [inline test/format/ops.c:421]: If we're thinking of merging this into develop sometime, we'll need to fix this change for predictable replay (the call to `replay_end_timed_run()` got lost).  Alternatively, these changes could be made with `#ifdef ENABLE_ANTITHESIS`

**Review by ddanderson — COMMENTED** [review]
A question/concern about changes to `format/ops.c`.

**Review by ddanderson — APPROVED** [review]
LGTM

**Pull request merged**